gpt_index/readers/weaviate/data_structs.py

```diff
@@ -250,3 +250,19 @@ def delete_document(cls, client: Any, ref_doc_id: str, class_prefix: str) -> Non
         entries = parsed_result[class_name]
         for entry in entries:
             client.data_object.delete(entry["_additional"]["id"], class_name)
+
+    @classmethod
+    def from_gpt_index_batch(
+        cls, client: Any, nodes: List[Node], class_prefix: str
+    ) -> List[str]:
+        """Convert from gpt index."""
+        from weaviate import Client  # noqa: F401
+
+        client = cast(Client, client)
+        validate_client(client)
+        index_ids = []
+        with client.batch as _:
```

Comment:
The context manager is actually needed.

Otherwise, the upload will use the default settings.

Reply:
also, when using the context manager, it is no longer necessary to flush the batch i.e. https://github.dev/jerryjliu/gpt_index/blob/e1bfb6ac6b05539464fb22a0ae667ac98fa42f2c/gpt_index/readers/weaviate/data_structs.py#L143-L144

Reply:
ohh i see. ok i'll use the batch from the context manager explicitly 

(also oops i left the flush call in, that was from earlier testing)